internal/controller/servermaintenanceset_controller_test.go

```diff
@@ -0,0 +1,172 @@
+		By("Creating a BMCSecret")
+		bmcSecret := &metalv1alpha1.BMCSecret{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "test-",
+			},
+			Data: map[string][]byte{
+				metalv1alpha1.BMCSecretUsernameKeyName: []byte("foo"),
+				metalv1alpha1.BMCSecretPasswordKeyName: []byte("bar"),
+			},
+		}
+		Expect(k8sClient.Create(ctx, bmcSecret)).To(Succeed())
+
+		By("Creating a Server")
+		server01 = &metalv1alpha1.Server{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "test-maintenance-",
+				Labels: map[string]string{
+					"metal.ironcore.dev/hostname": "test-hostname",
+				},
+			},
+			Spec: metalv1alpha1.ServerSpec{
+				UUID:       "38947555-7742-3448-3784-823347823834",
+				SystemUUID: "38947555-7742-3448-3784-823347823834",
+				BMC: &metalv1alpha1.BMCAccess{
+					Protocol: metalv1alpha1.Protocol{
+						Name: metalv1alpha1.ProtocolRedfishLocal,
+						Port: 8000,
+					},
+					Address: "127.0.0.1",
+					BMCSecretRef: v1.LocalObjectReference{
+						Name: bmcSecret.Name,
+					},
+				},
+			},
+		}
+		Expect(k8sClient.Create(ctx, server01)).Should(Succeed())
+
+		By("Creating a second Server")
+		server02 = &metalv1alpha1.Server{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "test-maintenance-",
+				Labels: map[string]string{
+					"metal.ironcore.dev/hostname": "test-hostname",
+				},
+			},
+			Spec: metalv1alpha1.ServerSpec{
+				UUID:       "38947555-7742-3448-3784-823347823834",
+				SystemUUID: "38947555-7742-3448-3784-823347823834",
+				BMC: &metalv1alpha1.BMCAccess{
+					Protocol: metalv1alpha1.Protocol{
+						Name: metalv1alpha1.ProtocolRedfishLocal,
+						Port: 8000,
+					},
+					Address: "127.0.0.1",
+					BMCSecretRef: v1.LocalObjectReference{
+						Name: bmcSecret.Name,
+					},
+				},
+			},
+		}
+		Expect(k8sClient.Create(ctx, server02)).Should(Succeed())
+	})
+
+	AfterEach(func(ctx SpecContext) {
+		DeleteAllMetalResources(ctx, ns.Name)
+	})
+	It("should successfully reconcile the resource", func(ctx SpecContext) {
+		By("Reconciling the created resource")
+
+		servermaintenanceset = &metalv1alpha1.ServerMaintenanceSet{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "test-maintenance-",
+				Namespace:    ns.Name,
+			},
+			Spec: metalv1alpha1.ServerMaintenanceSetSpec{
+				ServerSelector: metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"metal.ironcore.dev/hostname": "test-hostname",
+					},
+				},
+				Template: metalv1alpha1.ServerMaintenanceSpec{
+					Policy:    metalv1alpha1.ServerMaintenancePolicyEnforced,
+					ServerRef: &v1.LocalObjectReference{},
+				},
+			},
+		}
+		Expect(k8sClient.Create(ctx, servermaintenanceset)).To(Succeed())
+
+		By("Checking if the resource has been reconciled")
+
+		maintenance01 := &metalv1alpha1.ServerMaintenance{}
+
+		Eventually(k8sClient.Get).WithArguments(ctx, types.NamespacedName{
+			Name:      servermaintenanceset.Name + "-0",
+			Namespace: servermaintenanceset.Namespace,
+		}, maintenance01).Should(Succeed())
```

Comment:
As we are using `"sigs.k8s.io/controller-runtime/pkg/envtest/komega"` in all our tests we should consolidate it here and also e.g. use `Get()` or `Object()`. /ref https://github.com/ironcore-dev/metal-operator/blob/main/internal/controller/bmc_controller_test.go#L46-L69

Reply:
I would ensure that the fields also are matching whatever is put in the `Template` of the set.